database/bfgd/postgres/postgres.go

```diff
@@ -965,6 +965,57 @@ func (p *pgdb) AccessPublicKeyDelete(ctx context.Context, publicKey *bfgd.Access
 	return nil
 }
 
+// BtcBlocksHeightsWithNoChildren returns the heights of blocks stored in the
+// database that do not have any children, these represent possible forks that
+// have not been handled yet.
+func (p *pgdb) BtcBlocksHeightsWithNoChildren(ctx context.Context) ([]uint64, error) {
+	log.Tracef("BtcBlocksHeightsWithNoChildren")
+	defer log.Tracef("BtcBlocksHeightsWithNoChildren exit")
+
+	// Query all heights from btc_blocks where the block does not have any
+	// children and there are no other blocks at the same height with children.
+	// Excludes the tip because it will not have any children.
+	const q = `
+		SELECT height FROM btc_blocks bb1
+		WHERE NOT EXISTS (SELECT * FROM btc_blocks bb2 WHERE substr(bb2.header, 5, 32) = bb1.hash)
+		AND NOT EXISTS (
+			SELECT * FROM btc_blocks bb3 WHERE bb1.height = bb3.height 
+			AND EXISTS (
+				SELECT * FROM btc_blocks bb4 WHERE substr(bb4.header, 5, 32) = bb3.hash
+			)
+		)
+		ORDER BY height DESC
+		OFFSET $1 + 1
+		LIMIT 100
+	`
+
+	var heights []uint64
+	for offset := 0; ; offset += 100 {
```

Comment:
Does querying this data in groups of 100 have any impact on performance, or is it for another reason? 🤔

Reply:
meh it was just a way to implement pagination, the number was somewhat arbitrary; the last time I looked at testnet, we're actually at about 80 total blocks affected by this, my thinking was that we will probably never get more than 100 except for possible the first time we deploy this, so just paginate by that (and expect to not have more than a page after the first deploy, I'd guess we'd be returning 1 or 2 results but need to test)